### PR TITLE
Feature/FP-12

### PR DESCRIPTION
**Pull request by natam**
FP-12 Added unit/integration tests

### CODE REVIEW

File: src/test/java/com/nkh/ECommerceShop/security/jwt/JwtUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.util.WebUtils;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = JwtUtils.class)
```
Comment (faruka-startsteps) [inline src/test/java/com/nkh/ECommerceShop/security/jwt/JwtUtilsTest.java:17]: No need to initialize spring

File: src/test/java/com/nkh/ECommerceShop/controller/AuthControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+import java.time.Instant;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
```
Comment (faruka-startsteps) [inline src/test/java/com/nkh/ECommerceShop/controller/AuthControllerTest.java:30]: No need to initilize the whole spring framework. You can use the @WebMvcTest to init just the controllers and then you can provide mocks for your repositories

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/test/java/com/nkh/ECommerceShop/security/jwt/JwtUtilsTest.java (hunk 1 above)
Comment (natam) [reply src/test/java/com/nkh/ECommerceShop/security/jwt/JwtUtilsTest.java]: changed

File: src/test/java/com/nkh/ECommerceShop/controller/AuthControllerTest.java (hunk 2 above)
Comment (natam) [reply src/test/java/com/nkh/ECommerceShop/controller/AuthControllerTest.java]: changed

**Review by faruka-startsteps — APPROVED** [review]

**Pull request merged**